doc/api/net.md

```diff
@@ -515,56 +544,74 @@ changes:
     description: The `hints` option is supported now.
 -->
 
-Opens the connection for a given socket.
-
-For TCP sockets, `options` argument should be an object which specifies:
-
-  - `port`: Port the client should connect to (Required).
-
-  - `host`: Host the client should connect to. Defaults to `'localhost'`.
+* `options` {Object}
+* `connectListener` {Function} Common parameters of [`socket.connect()`][]
+  methods. Will be added as a listener for the [`'connect'`][] event once.
+* Returns: {net.Socket} The socket itself.
 
-  - `localAddress`: Local interface to bind to for network connections.
+Initiate the connection on a given socket.
 
-  - `localPort`: Local port to bind to for network connections.
+Note: this method is similar to a [connect(2)][] call, and before that,
+a [bind(2)][] if `localAddress` or `localPort` is specified,
+a [getaddrinfo(3)][] call if the `host` is specified but not an IP and `lookup`
+is left as default.
 
-  - `family` : Version of IP stack. Defaults to `4`.
+For TCP connections, available `options` are:
 
-  - `hints`: [`dns.lookup()` hints][]. Defaults to `0`.
+* `port` {number} Required. Port the socket should connect to.
+* `host` {string} Host the socket should connect to. Defaults to `'localhost'`.
+* `localAddress` {string} Optional. Local address the socket should send
+  connections from.
+* `localPort` {number} Optional. Local port the socket should send connections
+  from.
+* `family` {number}: Version of IP stack, can be either 4 or 6. Defaults to 4.
+* `hints` {number} Optional [`dns.lookup()` hints][].
+* `lookup` {Function} Custom lookup function. Defaults to [`dns.lookup()`][].
 
-  - `lookup` : Custom lookup function. Defaults to `dns.lookup`.
+For connections to local UNIX domain sockets or Windows named pipes, available
+`options` are:
 
-For local domain sockets, `options` argument should be an object which
-specifies:
+* `path` {string} Required. Path the client should connect to.
 
-  - `path`: Path the client should connect to (Required).
+#### socket.connect(path[, connectListener])
 
-Normally this method is not needed, as `net.createConnection` opens the
-socket. Use this only if you are implementing a custom Socket.
+* `path` {string} Required. Path the client should connect to.
```

Comment:
I  think "Required." is unnecessary, its clearly not optional in the syntax above. There would be an awful lot of "Required."  in our docs if we put that as the first word sentence for every mandatory positional parameter, which we don't.